File: app/src/main/java/com/moby/app/AppDatabase.java
package com.moby.app;

import android.content.Context;
import androidx.room.*;

/**
 * AppDatabase — Room Database.
 * TODO: Tambah Entity baru ke array entities = {User.class, ...}
 */
@Database(entities = {User.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase instance;

    public abstract UserDao userDao(); // TODO: tambah DAO lain di sini

    public static synchronized AppDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(
                context.getApplicationContext(), AppDatabase.class, "app_db")
                .fallbackToDestructiveMigration().build();
        }
        return instance;
    }
}

File: app/src/main/java/com/tes/jk/AppDatabase.kt
package com.tes.jk

import android.content.Context
import androidx.room.*

/**
 * AppDatabase — Room Database singleton.
 * TODO: Tambah Entity baru: entities = [User::class, ...]
 * PERHATIAN: fallbackToDestructiveMigration() menghapus data
 * saat schema berubah. Ganti dengan migrasi di production.
 */
@Database(entities = [User::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun userDao(): UserDao

    companion object {
        @Volatile private var instance: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase =
            instance ?: synchronized(this) {
                instance ?: Room.databaseBuilder(
                    context.applicationContext,
                    AppDatabase::class.java, "app_db"
                ).fallbackToDestructiveMigration().build().also { instance = it }
            }
    }
}
